lib/openstax/exchange/client/real_client.rb

```diff
@@ -0,0 +1,80 @@
+require 'oauth2'
+
+module OpenStax
+  module Exchange
+    module Client
+      class RealClient
+        include ClientInstance
+
+        def initialize(client_configuration)
+          @platform_id     = client_configuration.platform_id
+          @platform_secret = client_configuration.platform_secret
+          @server_base_url = client_configuration.server_base_url
+          @server_port     = client_configuration.server_port
+          @api_version     = client_configuration.api_version
+
+          @server_base = "#{@server_base_url}:#{@server_port}"
```

Comment:
Splitting the `url` and the `port` apart makes it more likely that someone will specify a trailing slash on the URL which will make the concatenated value wrong, e.g. `http://localhost/:3000`.  Unless there's a need I'm not seeing to split them, just have a base url where the caller has to provide the port if necessary.